src/VisualStudio/Core/Def/Implementation/InheritanceMargin/InheritanceGlyphFactory.cs

```diff
@@ -44,21 +51,36 @@ internal sealed class InheritanceGlyphFactory : IGlyphFactory
 
         public UIElement? GenerateGlyph(IWpfTextViewLine line, IGlyphTag tag)
         {
-            if (tag is not InheritanceMarginTag inheritanceMarginTag)
+            var document = _textView.TextBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document == null)
+            {
                 return null;
+            }
 
-            var membersOnLine = inheritanceMarginTag.MembersOnLine;
-            Contract.ThrowIfTrue(membersOnLine.IsEmpty);
+            var optionService = document.Project.Solution.Workspace.Services.GetRequiredService<IOptionService>();
```

Comment:
so we likely could simplify all of this if this isn't a language-specific option.  then, we an just map from TextBuffer to workspace, and check the workspace option here.